### ISSUE TITLE
Use url-caster.physical-web.org instead url-caster.appspot.com

### ISSUE THREAD

**Issue by beaufortfrancois** [google/physical-web#361]
For the sake of consistency and homogeneity, I would love if we would use https://url-caster.physical-web.org instead of https://url-caster.appspot.com.
What do you think? It would be as trivial as:
- Adding a Domain Name to GAE: https://appengine.google.com/deployment/newdomain?app_id=s~url-caster 
- Updating URLS in iOS & Android Apps and PWS to point to the new URLs.

**Comment by mmocny** [comment]
On the surface this seems like a reasonable request.  However, I think the physical-web.org domain should stay out of reference implementations, perhaps.

Specifically, the current PWS isn't official, its just someone one of our team members is running.  There may be several official PWS pop up over time (i.e. by Google, by other companies..), and we expect clients to choose their own backend as they do a search engine.

I think I'm negative on this change for that reason.  It would make it seem like its the one-true-PWS, which we don't want..

Either way, PWS is only ever interacted with by apps and tinkerers, so it wouldn't really get noticed.

**Comment by beaufortfrancois** [comment]
Let's close the issue then.
Thanks for your explanation Michal!

**Comment by mmocny** [comment]
Having slept on this, I'm not sure I agree with my own advice :P

physical-web.org is nothing more than a reference, but its still silly to be using an appspot domain when the reference apps depend on it for functionality.  I think we do communicate the state of the relationships between everything well enough in our docs.

I'll bring this up today and try to make the update.

**Comment by beaufortfrancois** [comment]
Any news?

**Comment by mmocny** [comment]
I don't have rights to do this, I'll bring it up today with Scott.

**Comment by beaufortfrancois** [comment]
(gentle ping)

**Comment by dinhviethoa** [comment]
We agree that it would be a nice to have (as long as the old URL still works).

**Comment by mmocny** [comment]
Working on it with Scott.  Trouble with some of the config, but will vc it
out this week.

**Comment by mmocny** [comment]
Woot, [service.physical-web.org](service.physical-web.org/demo) and [service-dev.physical-web.org](service-dev.physical-web.org/demo) are live!

See it in action:

`curl -k -s -H "Content-Type: application/json" -d '{"objects":[{"url":"http://www.google.com"}]}' http://service.physical-web.org/resolve-scan`

**Comment by scottjenson** [comment]
Very cool! Just noticed that the icon pointer is still using appspot:

 "icon": "https://url-caster.appspot.com/favicon?url=http...

Separate issue?

**Comment by mmocny** [comment]
It is, but thanks for raising it.

**Comment by mmocny** [comment]
Damn, in process of fixing this, I realize that https is broken: [service.physical-web.org](https://service.physical-web.org/demo) and [service-dev.physical-web.org](https://service-dev.physical-web.org/demo)!

Cannot update the url until those are fixed :(

**Comment by mmocny** [comment]
Scott can you take a scan at domains.google.org to see if we can fix this, maybe there is an option?

**Comment by dinhviethoa** [comment]
I guess you would need a SSL certificate (yay.)

**Comment by scottjenson** [comment]
Yeah, Matthew added HTTPS to PWDEMO.ORG and it was quite a process. He
could help on Tuesday.

Scott

**Comment by dermike** [comment]
Just found out there's a .pw domain that might just be the perfect fit. ;) 99c for the first year at http://www.buy.pw

**Comment by scottjenson** [comment]
demo.pw is only $500.... ;-(

**Comment by matthewsibigtroth** [comment]
I'm happy to help with the cert purchase. Installing on our node servers
was pretty straightforward, so hopefully the same applies to the pws
server.  :)

**Comment by mmocny** [comment]
Do we need a new cert for every domain, or every subdomain?  Perhaps we just need to set up the PWS with existing certs?

**Comment by mmocny** [comment]
Also, which node server?  Seems `physical-web.org` does a 302 redirect to `http://google.github.io/physical-web/`, which I assume is configured with google domains.  `https://physical-web.org` is invalid.

**Comment by dinhviethoa** [comment]
I think we don't even have a cert for physical-web.org.

**Comment by matthewsibigtroth** [comment]
We purchased a standard cert for pwdemo.org, which is our domain that we're
hosting our prototype/demo node servers on. Our cert does not support
subdomains, but one can pay more for a wildcard cert that will support all
subdomains, I believe.

**Comment by mmocny** [comment]
Makes sense, based on some quick reading, you have to purchase foo.com, *.
foo.com, *.*.foo.com etc.

**Comment by scottjenson** [comment]
Can we close this out yet?

**Comment by mmocny** [comment]
I think we still need to add support for `https` but that's not being actively pursued at the moment.
